### PR TITLE
Resolving Issue: Errors in Simple Handwritten PDEs as ODEs

### PR DESCRIPTION
**Pull request by GodotMisogi**
## Checklist

- [x] Appropriate tests were added
- [x] Any code changes were done in a way that does not break public API
- [x] All documentation related to code changes were updated
- [ ] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [x] Any new documentation only uses public API
  
## Additional context

This merges changes to resolve issue #1284 on the benchmarks of handwritten PDEs as ODEs via finite-difference and pseudo-spectral methods.

The package dependencies have been updated to remove unused ones, and the compat entries have been updated for Julia v1.10. The implementations have been tested on local machines (Macbook Pro M1 Pro on macOS Sequoia and a beefier CPU running Ubuntu 22.04 LTS) by running `SciMLBenchmarks.weave_file` on the relevant `.jmd` files.

At present, changes have been implemented for the following PDEs in one spatial and one time dimension: 

1. Allen-Cahn equation
2. Burgers' equation
3. Kortewreg-de Vries equation
4. Kuramoto-Sivashinsky equation

The following changes have been implemented to the finite-difference method benchmarks:

1. The implementation errors in the PDEs have been corrected; some have been updated to reflect their particular characteristics.
2. Sparse matrices are constructed using `SparseArrays`.
3. The `MatrixOperator` interfaces are used from the updated `SciMLOperators` for `SplitODEProblem`.
4. Sketches of the equations have been added (but require upcoming refinements and updates).
5. Space-time heatmaps of the solutions instead of line plots.

These are functioning for certain solver algorithm choices (`IMEXEuler`, `CNAB2`, `CLAF2`, `ETDRK2`, `SBDF2`) 

Commits will be progressively added in this pull (such as pseudo-spectral method updates), including revisions based on comments from the reviewers (@ChrisRackauckas).

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
For the psuedospectral, see the comment in https://github.com/SciML/SciMLBenchmarks.jl/issues/929. The ApproxFun dev mentioned we should change to OrthogonalPolynomials.jl for this part so we should take that recommendation.

**Comment by GodotMisogi** [comment]
@ChrisRackauckas Allen-Cahn and Burgers' equations via Chebyshev pseudospectral methods have been implemented using `ClassicalOrthogonalPolynomials.jl`. The Kuramoto-Sivashinsky and Korteweg-De Vries equations via Fourier pseudospectral methods are now working with `ApproxFun.jl`. These two cases will likely need to be reformulated to utilize Chebyshev interpolants with Neumann boundary conditions, allowing for the use of mass matrix solvers. 

Could you please run the workflow so I can check the logs for errors?

File: benchmarks/SimpleHandwrittenPDE/allen_cahn_spectral_wpd.jmd [hunk 1]
```diff
@@ -1,224 +1,123 @@
...
-plot(wp5, label=labels, markershape=:auto, title="ExpRK methods, medium order")
+abstols = 0.1 .^ (5:8) # all fixed dt methods so these don't matter much
+reltols = 0.1 .^ (1:4)
+multipliers = 0.5 .^ (0:3)
+setups = [
+    Dict(:alg => Rodas3(), :dts => 1e-4 * multipliers),
+    Dict(:alg => Rodas4(), :dts => 1e-4 * multipliers),
+    Dict(:alg => Rodas5(), :dts => 1e-4 * multipliers),
+    Dict(:alg => GRK4T(), :dts => 1e-4 * multipliers),
+    Dict(:alg => GRK4A(), :dts => 1e-4 * multipliers),
```
Comment (ChrisRackauckas) [inline benchmarks/SimpleHandwrittenPDE/allen_cahn_spectral_wpd.jmd:102]: the main ones to add to the mass matrices comparisons would be FBDF and QNDF. Also a DAEProblem formulation testing DASKR.jl and Sundials.IDA. You can benchmark multiple simultaneous problem implementations using the `prob_choice` setup, see the DAE benchmarks for how this is done.

Comment (GodotMisogi) [reply benchmarks/SimpleHandwrittenPDE/allen_cahn_spectral_wpd.jmd:102]: Great to know; thanks for the tips!

**Comment by GodotMisogi** [comment]
@ChrisRackauckas Please run the workflow so I can check the logs for errors?

1. The problem descriptions, including equations, for all partial differential equation benchmark tests have been added.
2. Allen-Cahn and Burgers' equations via finite-difference methods have been implemented using [`SummationByPartsOperators.jl`](https://github.com/ranocha/SummationByPartsOperators.jl/tree/main) for conveniently specifying the accuracy of the derivative operator representations.
3. Allen-Cahn and Burgers' equations via Chebyshev pseudo-spectral methods have been implemented using [`ClassicalOrthogonalPolynomials.jl`](github.com/JuliaApproximation/ClassicalOrthogonalPolynomials.jl). The use of non-singular mass matrix solvers has been side-stepped by implementing LU factorization of the mass matrix and inverting the linear systems in the vector field implementations provided to `SplitODEProblem`.
4. The Kuramoto-Sivashinsky and Korteweg-De Vries equations via finite-difference and Fourier pseudo-spectral methods are implemented using `SummationByPartsOperators.jl` with periodic boundary conditions.
5. All tests have included the previous sets of benchmarks comparing implicit-explicit and exponential integrator methods, along with comparisons between the families. Some solvers have been excluded due to excessively long runtimes, which will need further diagnostics.

File: benchmarks/SimpleHandwrittenPDE/kdv_fdm_wpd.jmd [hunk 2]
```diff
@@ -1,245 +1,187 @@
...
-                            save_everystep=false, appxsol=test_sol, maxiters=Int(1e5));
-
-plot(wp, label=labels, markershape=:auto, title="Between family, medium order")
+abstols = 0.1 .^ (5:8) # all fixed dt methods so these don't matter much
+reltols = 0.1 .^ (1:4)
+multipliers = 0.5 .^ (0:3)
+setups = [
+    Dict(:alg => CNAB2(), :dts => 1e-4 * multipliers),
+    Dict(:alg => CNAB2(linsolve=KrylovJL_GMRES()), :dts => 1e-4 * multipliers),
+    Dict(:alg => ETDRK2(), :dts => 1e-4 * multipliers),
```
Comment (ChrisRackauckas) [inline benchmarks/SimpleHandwrittenPDE/kdv_fdm_wpd.jmd:173]: This is missing all of the good high order methods?

Comment (GodotMisogi) [reply benchmarks/SimpleHandwrittenPDE/kdv_fdm_wpd.jmd:259]: @ChrisRackauckas Please run the workflow on the latest commit? I think it's mostly ready.
